deebot_client/commands/json/map.py

```diff
@@ -246,6 +269,93 @@ def _handle_body_data_dict(
         return HandlingResult.analyse()
 
 
+class GetMapSetV2(JsonCommandWithMessageHandling, OnMapSetV2):
+    """Get map set v2 command."""
+
+    name = "getMapSet_V2"
+
+    def __init__(
+        self,
+        mid: str,
+        type: (  # pylint: disable=redefined-builtin
+            MapSetType | str
+        ) = MapSetType.ROOMS,
+    ) -> None:
+        if isinstance(type, MapSetType):
+            type = type.value
+
+        super().__init__({"mid": mid, "type": type})
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        # check if type is know and subset us given
+        if not MapSetType.has_value(data["type"]) or not data.get("subsets"):
+            return HandlingResult.analyse()
+
+        # subset is based64 7z compressed
+        subsets: list[list[str]] = json.loads(
+            decompress_7z_base64_data(data["subsets"]).decode()
+        )
+
+        # handle rooms
+        if data["type"] in (MapSetType.ROOMS):
+            room_subsets: list[dict[str, Any]] = [
+                {
+                    "id": int(subset[0]),  # room id
+                    "name": subset[1]
+                    if subset[1] and subset[1] != " "
+                    else "Default",  # room name
+                    # subset[2] not sure what the value is for
+                    # subset[3] not sure what the value is for
+                    # subset[4] room clean order
+                    "coordinates": f"{subset[5]},{subset[6]}",  # room center coordinates
```

Comment:
The coordinates on `GetMapSet` are the points/border of the specific room. These coordinates can be used to determine if the bot is in a specific room (https://github.com/DeebotUniverse/client.py/pull/397).

Does the V2 also have these values? At least, it looks like if you check out the PR.
We can use the center coordinates later in a follow up PR

Reply:
yes, the coordinate is the center of a room for newer bots.
For older bots i saw they use points of the room boarder.

You could not check for new bots this way if they are in a specific room as we only have the center of the room now.
For newer bots we get info for different room over the image, as they have now possibility for different coloring. It can be colored different by add colors in numbers higher than 100 inside:
- https://github.com/MVladislav/client.py/blob/9d203bab078b1fa499fd8798e40d6803a226445a/deebot_client/map.py#L127
- https://github.com/MVladislav/client.py/blob/9d203bab078b1fa499fd8798e40d6803a226445a/deebot_client/map.py#L346

I have had a test for set a room title in the map center, what was tested here: https://github.com/MVladislav/client.py/blob/cd47c6f4d1fbf712186ec4e1beda9d815bd813e9/deebot_client/map.py#L390